src/tests/ftest/launch.py

```diff
@@ -897,6 +901,74 @@ def archive_logs(avocado_logs_dir, test_yaml, args):
     spawn_commands(host_list, "; ".join(commands), 900)
 
 
+def send_notification(hosts, subject, msg, attachment, emails):
+    """Send email notification to provided emails with given message.
+
+    Args:
+        host (list): hosts to perform command in.
+        msg (str): message to be sent.
+        attachment (str): attachment to send on the email.
+        emails (list): list of email addresses to send message to.
+    """
+    mail_cmd = "mail -s \"{}\" -a {} {} <<< \"{}\"".format(
+        subject, attachment, ",".join(emails), msg)
+    spawn_commands(hosts, mail_cmd, 30)
+
+
+def get_log_size(test_yaml, test_file, args, size_limit=2**33):
+    """Get the size of the the directoy of the host test log files in
+    the avocado results directory and store the values in a file.
+
+    Args:
+        test_yaml (str): yaml file containing host names
+        test_file (str): the test python file
+        args (argparse.Namespace): command line arguments for this program
+        size_limit (int): limitsize in bytes that determines when to send
+            notification.
+    """
+    logs_dir = os.environ.get("DAOS_TEST_LOG_DIR", DEFAULT_DAOS_TEST_LOG_DIR)
+    log_size_file = os.path.join(logs_dir, "log_size.log")
+    host_list = get_hosts_from_yaml(test_yaml, args)
+
+    # Create a file that contains the sizes on each test log and the logs_dir.
+    command = "du -ab -d 1 {} &> {}".format(logs_dir, log_size_file)
+    spawn_commands(host_list, command, 30)
+
+    # Check log_size.log file to see if anything goes over size_limit
+    size_info_cmd = "cat {}".format(log_size_file)
+    size_info_out = get_remote_output(host_list, size_info_cmd, 30)
+
+    # Create a dictionary of hosts for each unique return code
+    results = {code: hosts for code, hosts in size_info_out.iter_retcodes()}
+
+    # Determine if the command completed successfully across all the hosts
+    status = len(results) == 1 and 0 in results
+    if not status:
+        print("  Errors detected running \"{}\":".format(command))
+
+    # Check the command output
+    for code in sorted(results):
+        output_data = list(size_info_out.iter_buffers(results[code]))
+        print("Out data: {}".format(output_data))
+        if not output_data:
+            err_nodes = NodeSet.fromlist(results[code])
+            print("    {}: rc={}, output: <NONE>".format(err_nodes, code))
+        else:
+            msg = []
+            for output, o_hosts in output_data:
+                for line in str(output).splitlines():
+                    if line.split("\t")[0] and line.split("\t")[0].isdigit():
+                        if int(line.split("\t")[0]) > size_limit:
+                            msg.append("Host: {} \nTest File: {}\n".format(
+                                ",".join(o_hosts), test_file))
+                            break
+            if msg:
```

Comment:
Shouldn't `msg` have some indication or trail back to the test run where the big logs were found?

It seems currently you are going to get these e-mails but not have any way to go examine the test run that produced them.  Unless I am missing something.

Reply:
Ah yes, I need to get rid of the break.  But the message is populated with the o_hosts which is the hosts from which the output is found and the message also includes the test_file. So both of these will tell the recipient of the email which test the logs are coming from and which hosts the big logs were found in.

Reply:
But not which test run (i.e. PR)  What if, like me, others typically have a number of PRs in flight at the same time, typically due to test hardware scheduling latency.  I.e. working on one while waiting for another for any number of PRs.

Reply:
I see, so you mean the PR where the big logs where found? Along also with the PR build number?
For example the message could look like this: 
```
PR: 1234 Build: 10
Host: wolf-110
Test File: rebuild_test_1234
```
From looking at the code in launch.py I see that we get the 'avocado_logs_dir' variable and it stores the following path: /var/lib/jenkins/jenkins-1/docker_1/workspace/daos-stack_daos_PR-1989@3/install/lib/daos/TESTING/ftest/avocado/job-results/

I could parse this variable and get the PR and build info to pass to the message.

Reply:
A link to a test run would be ideal.

You should examine the environment variables set by Jenkins with `env`.  Most likely there is a URL in there or enough information there to construct a URL.

Reply:
Ok, I made a change to use the environment variable: BUILD_URL that I found here: https://build.hpdd.intel.com/env-vars.html/

Reply:
Why do you need human_to_bytes?  Why not just have the du command give you the results in bytes?

Does our test harness have a RPC style connection to to the nodes being tested so that it can get this type of information with out spawning shell commands to each node?

Reply:
> Does our test harness have a RPC style connection to to the nodes being tested so that it can get this type of information with out spawning shell commands to each node?

No.